api/utils/keys/yubikey.go

```diff
@@ -431,3 +449,37 @@ func selfSignedTeleportClientCertificate(priv crypto.PrivateKey, pub crypto.Publ
 	}
 	return cert, nil
 }
+
+// YubiKeys require touch when generating a private key that requires touch, or using
+// a private key (Sign) with touch required. Unfortunately, there is no good way to
+// check whether touch is cached by the PIV module at a given time. In order to require
+// touch only when needed, we prompt for touch after a short delay when we expect the
+// request would succeed if touch were not required.
+//
+// There are some X factors which determine how long a request may take, such as the
+// YubiKey model and firmware version, so the delays below may need to be adjusted to
+// suit more models. The durations mentioned below were retrieved from testing with a
+// YubiKey 5 nano (5.2.7) and a YubiKey NFC (5.4.3).
+const (
+	// piv.ECDSAPrivateKey.Sign consistently takes ~70 milliseconds. We don't want to delay signatures
```

Comment:
Are you sure that these timings are accurate? Nothing can influence them like different OS, different YK, etc?